x/application/keeper/query_application.go

```diff
@@ -25,7 +28,8 @@ func (k Keeper) AllApplications(ctx context.Context, req *types.QueryAllApplicat
 	pageRes, err := query.Paginate(applicationStore, req.Pagination, func(key []byte, value []byte) error {
 		var application types.Application
 		if err := k.cdc.Unmarshal(value, &application); err != nil {
-			return err
+			logger.Error(fmt.Sprintf("unmarshaling application with key (hex): %x: %+v", key, err))
+			return status.Error(codes.Internal, err.Error())
```

Comment:
Given all `Internal` errors were wrapping the original one, shouldn't this one do the same?

Reply:
`status` errors should only be returned to the gRPC client; we don't need to be using them internally. You caught a few places where, somehow, I introduced an inconsistency and was wrapping some errors too early. Those cases have been resolved and the rest should look like this already.